### PR TITLE
Outline BDTEventWrapper::GetVarIndex() to help lld.

### PR DESCRIPTION
**Pull request by Axel-Naumann**
ldd has problems with linking tmva due to this function in the header. Outlining it fixes it.

```
/usr/bin/ld: error: can't create dynamic relocation R_X86_64_TLSLD against symbol: TMVA::BDTEventWrapper::GetVarIndex()::fVarIndex in readonly segment; recompile object files with -fPIC
>>> defined in CMakeFiles/TMVA.dir/src/DecisionTree.cxx.o
>>> referenced by BDTEventWrapper.h:64 (/home/axel/build/root/branches/v6-10/cmake/include/TMVA/BDTEventWrapper.h:64)
>>>               CMakeFiles/TMVA.dir/src/DecisionTree.cxx.o:(void std::__introsort_loop<__gnu_cxx::__normal_iterator<TMVA::BDTEventWrapper*, std::vector<TMVA::BDTEventWrapper, std::allocator<TMVA::BDTEventWrapper> > >, long, __gnu_cxx::__ops::_Iter_less_iter>(__gnu_cxx::__normal_iterator<TMVA::BDTEventWrapper*, std::vector<TMVA::BDTEventWrapper, std::allocator<TMVA::BDTEventWrapper> > >, __gnu_cxx::__normal_iterator<TMVA::BDTEventWrapper*, std::vector<TMVA::BDTEventWrapper, std::allocator<TMVA::BDTEventWrapper> > >, long, __gnu_cxx::__ops::_Iter_less_iter) (.isra.194))
```

While it'd be great to submit a reproducer to lld (I didn't manage to create one within the few minutes I invested), this change might be a cheap workaround. Unless it's a hot function where outlining is a real cost?

### CODE REVIEW

File: tmva/tmva/src/BDTEventWrapper.cxx [hunk 1]
```diff
@@ -65,3 +65,13 @@ Double_t BDTEventWrapper::GetCumulativeWeight(Bool_t type) const {
    if(type) return fSigWeight;
    return fBkgWeight;
 }
+
+////////////////////////////////////////////////////////////////////////////////
+/// Index of the variable to sort on
+
+Int_t& BDTEventWrapper::GetVarIndex() {
+   // This is a workaround for OSx where static thread_local data members are
```
Comment (pcanal) [inline tmva/tmva/src/BDTEventWrapper.cxx]: Anyway, for order of initialization reason.  A function static is usually preferable to a class static.

**Comment by Axel-Naumann** [comment]
This is actually due to a GCC bug https://gcc.gnu.org/bugzilla/show_bug.cgi?id=82906 - lld does "the right thing". I'll compare the stressTMVA time with and without this patch before merging.

**Comment by vgvassilev** [comment]
ping...

**Comment by lmoneta** [comment]
@Axel-Naumann are we still having a problem in ldd for this one ? 
I would prefer to close this and not merge, since I suspect outlining this function will be very costly. It is heavily used in the event loop by the DecisionTree class.

**Pull request closed without merging**